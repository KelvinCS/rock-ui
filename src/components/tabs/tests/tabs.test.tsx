import { render } from '@testing-library/react';
import * as React from 'react';
import { TabList, Tab, Tabs, TabPanel, TabPanels } from '../';

test('Tabs renders correctly', () => {
  const { asFragment } = render(
    <Tabs>
      <TabList>
        <Tab>One</Tab>
        <Tab>Two</Tab>
      </TabList>

      <TabPanels>
        <TabPanel>
          <p>one!</p>
        </TabPanel>
        <TabPanel>
          <p>two!</p>
        </TabPanel>
      </TabPanels>
    </Tabs>
  );
  expect(asFragment()).toMatchSnapshot();
});
