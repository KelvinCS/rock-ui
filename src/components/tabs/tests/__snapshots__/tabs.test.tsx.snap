// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-tabs css-0"
  >
    <div
      aria-orientation="horizontal"
      class="chakra-tabs__tablist css-k008qs"
      role="tablist"
    >
      <button
        aria-controls="tabs-1--tabpanel-0"
        aria-disabled="false"
        aria-selected="true"
        class="chakra-tabs__tab css-u93537"
        id="tabs-1--tab-0"
        role="tab"
        tabindex="0"
        type="button"
      >
        One
      </button>
      <button
        aria-controls="tabs-1--tabpanel-1"
        aria-disabled="false"
        class="chakra-tabs__tab css-u93537"
        id="tabs-1--tab-1"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Two
      </button>
    </div>
    <div
      class="chakra-tabs__tab-panels css-0"
    >
      <div
        class="chakra-tabs__tab-panel css-0"
        id="tabs-1--tabpanel-0"
        role="tabpanel"
      >
        <p>
          one!
        </p>
      </div>
      <div
        class="chakra-tabs__tab-panel css-0"
        hidden=""
        id="tabs-1--tabpanel-1"
        role="tabpanel"
      >
        <p>
          two!
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
